### PR TITLE
fix(csi): prevent race condition in ci tests step

### PR DESCRIPTION
**Pull request by Al-Pragliola**
## Description

fixes a race condition found in the ci CSI tests: https://github.com/opendatahub-io/model-registry/actions/runs/10997757515/job/30534062040#logs

## How Has This Been Tested?

run the ci

## Merge criteria:

- All the commits have been [_signed-off_](https://github.com/kubeflow/community/tree/master/dco-signoff-hook#signing-off-commits)  (To pass the `DCO` check)

- [ ] The commits have meaningful messages; the author will squash them after approval or in case of manual merges will ask to merge with squash.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work.
- [x] Code changes follow the [kubeflow contribution guidelines](https://www.kubeflow.org/docs/about/contributing/).

### CODE REVIEW

**Comment by Al-Pragliola** [comment]
cc @lampajr

**Review by tarilabs — COMMENTED** [review]
thank you @Al-Pragliola this is great and cc @lampajr for awareness and in case this "pattern" of the sleep is used in other places?

/lgtm

File: csi/test/test_utils.sh
Comment (lampajr) [inline csi/test/test_utils.sh:1]: Great thing to add a separated `test_utils.sh`, do you mind to move https://github.com/kubeflow/model-registry/blob/e0598fd4d0d0a3d5671234ff7baee2a2460acd54/csi/test/e2e_test.sh#L9-L15 here as well?

**Review by lampajr — COMMENTED** [review]
Thanks a lot @Al-Pragliola, it looks great to me!!!

@tarilabs the only other place where we could use the same approach is here https://github.com/kubeflow/model-registry/blob/e0598fd4d0d0a3d5671234ff7baee2a2460acd54/csi/test/e2e_test.sh#L171-L175

iirc I introduced the 5s sleep because sometimes the `res` was not populated.. but I don't rembember if it was failing in retrieving `SERVICE_HOSTNAME` or the `res` directly. So I think this could be done as a follow-up PR if needed.

File: csi/test/test_utils.sh
Comment (Al-Pragliola) [reply csi/test/test_utils.sh:1]: done

**Review by lampajr — APPROVED** [review]
Awesome, thanks @Al-Pragliola !

/lgtm
/approve

**Review by tarilabs — APPROVED** [review]
- https://github.com/kubeflow/model-registry/pull/426#pullrequestreview-2325908271
- https://github.com/kubeflow/model-registry/pull/426#discussion_r1774968146
- https://github.com/kubeflow/model-registry/pull/426#pullrequestreview-2328116950

/lgtm
/approve

and many thanks again @Al-Pragliola

**Pull request merged**